test/engine.test.ts

```diff
@@ -125,6 +130,35 @@ const sports1xConditionLe: EngineCondition = {
 const vectorCondition: EngineCondition = {
   filter: {
     any: false,
+    version: 1,
+    queries: [
+      {
+        property: 'topicDist',
+        filterComparisonType: 'vectorDistance',
+        value: {
+          vector: [0.4, 0.8, 0.3],
+          threshold: 0.5,
+        },
+      },
+    ],
+  },
+  rules: [
+    {
+      reducer: {
+        name: 'count',
+      },
+      matcher: {
+        name: 'ge',
+        args: 1,
+      },
+    },
+  ],
+};
+
+const vectorConditionV2: EngineCondition = {
+  filter: {
+    any: false,
+    version: 2,
```

Comment:
We set feature version at the feature level, such as [here](https://github.com/AirGrid/edgekit/pull/59/files#diff-30dc11e8f9af9a694581f0a9169aa97fR205), but the query for it at the audience level - any reason why over setting inside each of the `queries` which targets a `property` and that property is what is actually versioned.

Reply:
I was thinking it should be higher up because it needs to correspond to the audience definition version. Right now there's no support for multiple query objects but the query property is still an array, so it seems like it might end up just being redundant. If the audience definition changes to support more than one query object then it would also have more versions, WDYT?